Sources/FoundationInternationalization/Formatting/Number/BinaryInteger+FormatStyle.swift

```diff
@@ -43,250 +43,183 @@ extension BinaryInteger {
-        //
-        //   2. It returns the size of the value in two's complement representation, which includes the sign bit that we don't care about.  But we can't just subtract one from that value [for signed types], because for negative powers of two the signed form is one bit shorter than the [unsigned] magnitude's (signed types can represent -(2^N) through +(2^N)-1 - note the asymmetry).  That special case can technically be handled, but it requires determining if `self` is a negative power of two which is relatively expensive for large `BinaryInteger`s.
-        //
-        //   3. Some `BinaryInteger` implementations implement `bitWidth` wrong, due to its terse and ambiguous documentation.  e.g. some are also `FixedWidthInteger`s yet _don't_ return the maximum bit width of the type (or vice versa), some have off-by-one errors for negative powers of two, etc.  Although it's not this code's responsibility to allow for implementation errors, it's nice to.
-        //
-        // So, it's both necessary and preferable (respectively) to just examine `words` directly.
-
-        if .zero == self {
-            return 1
-        } else if .zero <= self {
-            // Find the highest-order word with any bits set, determine the overall index of the highest set bit, and return that plus one (index to count conversion).
-
-            for (i, word) in words.reversed().enumerated() {
-                if .zero != word {
-                    let fullWidth = type(of: word).bitWidth
-                    return ((words.count - i - 1) * fullWidth) + (fullWidth - word.leadingZeroBitCount)
-                }
-            }
-
-            preconditionFailure("\(type(of: self)) \(self) compared as not zero yet all its words are zero.")
-        } else { // `self` is negative.
-            // Perform two's complement one word at a time, keeping track of the index of the highest set bit seen so far.  After enumerating all the words, return that index plus one (index to count conversion).
-
-            var carryingOne = true // Covers both the initial +1 (as part of two's complement) and overflow between words.
-            var indexOfHighestSetBitSeenSoFar = 0
-
-            for (i, word) in words.enumerated() {
-                var positiveWord = ~word
-
-                if carryingOne {
-                    (positiveWord, carryingOne) = positiveWord.addingReportingOverflow(1)
-                }
-
-                let fullWidth = type(of: word).bitWidth
-
-                if carryingOne {
-                    indexOfHighestSetBitSeenSoFar += fullWidth
-                } else if .zero != positiveWord {
-                    indexOfHighestSetBitSeenSoFar = (fullWidth * i) + (fullWidth - positiveWord.leadingZeroBitCount) - 1
-                }
-            }
-
-            return indexOfHighestSetBitSeenSoFar + 1
+    let radix: (exponent: Int, power: UInt) = maxDecimalExponentAndPowerForUnsignedIntegerWord()
```

Comment:
It might be better to fetch that in the closure below, as it appears the compiler doesn't automatically move it there, so this is incurring unnecessary closure argument passing overhead.

(which is minor - some wasted stack space and register juggling overhead, mainly - but might be precluding further optimisations through inlining)